autodispose-android/src/main/java/com/uber/autodispose/android/LifecycleOwnerScopeProvider.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2017. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.autodispose.android;
+
+import android.arch.lifecycle.Lifecycle;
+import android.arch.lifecycle.LifecycleOwner;
+import com.uber.autodispose.LifecycleEndedException;
+import com.uber.autodispose.LifecycleNotStartedException;
+import com.uber.autodispose.LifecycleScopeProvider;
+import io.reactivex.Observable;
+import io.reactivex.functions.Function;
+
+import static android.arch.lifecycle.Lifecycle.State.DESTROYED;
+
+/**
+ * A {@link LifecycleScopeProvider} that can provide scoping for Android
+ * {@link LifecycleOwner} classes.
+ * <p>
+ * <pre><code>
+ *   AutoDispose.observable()
+ *      .scopeWith(LifecycleOwnerScopeProvider.from(view))
+ *      .empty();
+ * </code></pre>
+ */
+public final class LifecycleOwnerScopeProvider implements LifecycleScopeProvider<Lifecycle.State> {
+
+  private static final Function<Lifecycle.State, Lifecycle.State> CORRESPONDING_EVENTS =
+      new Function<Lifecycle.State, Lifecycle.State>() {
+        @Override public Lifecycle.State apply(Lifecycle.State lastEvent) throws Exception {
+          switch (lastEvent) {
+            case INITIALIZED:
+              throw new LifecycleNotStartedException();
+            case CREATED:
```

Comment:
I think you are confusing states vs events. 
https://developer.android.com/topic/libraries/architecture/lifecycle.html

e.g. both onStart and onPause brings you to the STARTED state. I'm not sure where you are using these but this seems wrong.